src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -0,0 +1,372 @@
+  useEffect(() => {
+    isMounted.current = true;
+    loadSources();
+    const getTitles = async () => {
+      const indexPatternTitles = await indexPatternService.getTitles();
+      if (isMounted.current) {
+        setExistingIndexPatterns(indexPatternTitles);
+        setIsLoadingIndexPatterns(false);
+      }
+    };
+    getTitles();
+    return () => {
+      isMounted.current = false;
+    };
+  }, [http, indexPatternService, loadSources]);
+
+  // loading rollup info
+  useEffect(() => {
+    const getRollups = async () => {
+      try {
+        const response = await http.get('/api/rollup/indices');
+        if (isMounted.current) {
+          setRollupIndicesCapabilities(response || {});
+        }
+      } catch (e) {
+        // Silently swallow failure responses such as expired trials
+      }
+    };
+
+    getRollups();
+  }, [http, type]);
+
+  const getRollupIndices = (rollupCaps: RollupIndicesCapsResponse) => Object.keys(rollupCaps);
+
+  const loadTimestampFieldOptions = useCallback(
+    async (query: string) => {
+      let timestampOptions: TimestampOption[] = [];
+      const isValidResult =
+        !existingIndexPatterns.includes(query) && matchedIndices.exactMatchedIndices.length > 0;
+      if (isValidResult) {
+        setIsLoadingTimestampFields(true);
+        const getFieldsOptions: GetFieldsOptions = {
+          pattern: query,
+        };
+        if (type === INDEX_PATTERN_TYPE.ROLLUP) {
+          getFieldsOptions.type = INDEX_PATTERN_TYPE.ROLLUP;
+          getFieldsOptions.rollupIndex = rollupIndex;
+        }
+
+        const fields = await ensureMinimumTime(
+          indexPatternService.getFieldsForWildcard(getFieldsOptions)
+        );
+        timestampOptions = extractTimeFields(fields, requireTimestampField);
+      }
+      if (isMounted.current) {
+        setIsLoadingTimestampFields(false);
+        setTimestampFieldOptions(timestampOptions);
+      }
+      return timestampOptions;
+    },
+    [
+      existingIndexPatterns,
+      indexPatternService,
+      requireTimestampField,
+      rollupIndex,
+      type,
+      matchedIndices.exactMatchedIndices,
+    ]
+  );
+
+  useEffect(() => {
+    loadTimestampFieldOptions(title);
+    getFields().timestampField?.setValue('');
+  }, [matchedIndices, loadTimestampFieldOptions, title, getFields]);
+
+  const reloadMatchedIndices = useCallback(
+    async (newTitle: string) => {
+      const isRollupIndex = (indexName: string) =>
+        getRollupIndices(rollupIndicesCapabilities).includes(indexName);
+      let newRollupIndexName: string | undefined;
+
+      const fetchIndices = async (query: string = '') => {
+        setIsLoadingMatchedIndices(true);
+        const indexRequests = [];
+
+        if (query?.endsWith('*')) {
+          const exactMatchedQuery = getIndices(http, isRollupIndex, query, allowHidden);
+          indexRequests.push(exactMatchedQuery);
+          indexRequests.push(Promise.resolve([]));
```

Comment:
Is this line necessary?

Reply:
I added a comment to the code. Its a default value when we're not making a request for the `partialMatchQuery`